src/goto-instrument/accelerate/acceleration_utils.cpp

```diff
@@ -633,10 +633,10 @@ bool acceleration_utilst::do_arrays(
 
   implies_exprt implies(k_bound, arrays_expr);
 
-  exprt forall(ID_forall);
+  forall_exprt forall;
   forall.type()=bool_typet();
-  forall.copy_to_operands(k);
```

Comment:
forall is always boolean, so I'd be tempted to tweak the class definition of forall_exprt to derive from binary_predicate_exprt instead of binary_exprt.

Reply:
Oh, and also to add a constructor that takes two arguments.